src/arith/rewrite_simplify.cc

```diff
@@ -460,7 +460,8 @@ PrimExpr RewriteSimplifier::Impl::VisitExpr_(const DivNode* op) {
 
   // x / 2.0 = x * 0.5
   if (const FloatImmNode* ptr = op->b.as<FloatImmNode>()) {
-    CHECK(op->dtype.is_float());
+    // TODO(@gussmith23) is this ok?
+    // CHECK(op->dtype.is_float());
```

Comment:
why do we need to do this @gussmith23 ?

Reply:
i think a better idea may be to do 
```CHECK(op->dtype.is_float() || datatype::Registry::Global()->GetTypeRegistered(op->dtype.code()));```

Reply:
also, it seems FloatImm(custom_datatype, value) represents a custom_datatype with that float value. see here: https://github.com/gussmith23/tvm/blob/a45f7bb1975933118db7647261a6ddefb214595a/include/tvm/tir/op.h#L829, we freely mutate FloatImm as a double